### ISSUE TITLE
Occasional exception in TrackingCollection.

### ISSUE THREAD

**Issue by grokys** [github/VisualStudio#597]
I get very occasional exceptions in TrackingCollection - usually when the PR list is updating. Here's the details:

```
At TrackingCollection.cs line 476

System.ObjectDisposedException occurred
  HResult=-2146232798
  Message=Cannot access a disposed object.
Object name: 'TrackingCollection'.
  ObjectName=TrackingCollection
  Source=GitHub.Exports.Reactive
  StackTrace:
       at GitHub.Collections.TrackingCollection`1.RemoveItem(T item) in D:\projects\VisualStudio\src\GitHub.Exports.Reactive\Collections\TrackingCollection.cs:line 476
  InnerException: 
```

As you can see, the stack trace shows only the current method: `RemoveItem`.

This crashes VS.

**Comment by shana** [comment]
Feels like a race condition between disposal of the current listeners in preparation for processing the new results and the current list still processing.

**Comment by StanleyGoldman** [comment]
I tried my best to reproduce this bug or see how it could happen and I checked the master and maintainer-workflow-diff branches. The only usages of `RemoveItem()` I could find are wrapped in `try/catch` blocks that swallow exceptions of type `ObjectDisposedException`. I'm not sure how you are getting that exception.

**Comment by grokys** [comment]
Just got this again with the following stack trace:

```
GitHub.Exports.Reactive.dll!GitHub.Collections.TrackingCollection<GitHub.Models.IPullRequestModel>.RemoveItem(GitHub.Models.IPullRequestModel item) Line 476	C#
GitHub.App.dll!GitHub.Services.ModelService.GetPullRequests.AnonymousMethod__4(GitHub.Services.ModelService.PullRequestCacheItem item) Line 168	C#
GitHub.App.dll!GitHub.Extensions.AkavacheExtensions.GetAndFetchLatestFromIndex.AnonymousMethod__13() Line 238	C#
mscorlib.dll!System.Runtime.CompilerServices.AsyncMethodBuilderCore.MoveNextRunner.InvokeMoveNext(object stateMachine)	Unknown
mscorlib.dll!System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx)	Unknown
mscorlib.dll!System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx)	Unknown
mscorlib.dll!System.Runtime.CompilerServices.AsyncMethodBuilderCore.MoveNextRunner.Run()	Unknown
mscorlib.dll!System.Runtime.CompilerServices.AsyncMethodBuilderCore.OutputAsyncCausalityEvents.AnonymousMethod__0()	Unknown
mscorlib.dll!System.Runtime.CompilerServices.AsyncMethodBuilderCore.ContinuationWrapper.Invoke()	Unknown
System.Reactive.Linq.dll!System.Reactive.Subjects.AsyncSubject<System.__Canon>.AwaitObserver.InvokeOnOriginalContext.AnonymousMethod__6_0(object c)	Unknown
WindowsBase.dll!System.Windows.Threading.ExceptionWrapper.InternalRealCall(System.Delegate callback, object args, int numArgs)	Unknown
WindowsBase.dll!System.Windows.Threading.ExceptionWrapper.TryCatchWhen(object source, System.Delegate callback, object args, int numArgs, System.Delegate catchHandler)	Unknown
WindowsBase.dll!System.Windows.Threading.DispatcherOperation.InvokeImpl()	Unknown
WindowsBase.dll!System.Windows.Threading.DispatcherOperation.InvokeInSecurityContext(object state)	Unknown
mscorlib.dll!System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx)	Unknown
mscorlib.dll!System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx)	Unknown
mscorlib.dll!System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state)	Unknown
WindowsBase.dll!MS.Internal.CulturePreservingExecutionContext.Run(MS.Internal.CulturePreservingExecutionContext executionContext, System.Threading.ContextCallback callback, object state)	Unknown
WindowsBase.dll!System.Windows.Threading.DispatcherOperation.Invoke()	Unknown
WindowsBase.dll!System.Windows.Threading.Dispatcher.ProcessQueue()	Unknown
WindowsBase.dll!System.Windows.Threading.Dispatcher.WndProcHook(System.IntPtr hwnd, int msg, System.IntPtr wParam, System.IntPtr lParam, ref bool handled)	Unknown
WindowsBase.dll!MS.Win32.HwndWrapper.WndProc(System.IntPtr hwnd, int msg, System.IntPtr wParam, System.IntPtr lParam, ref bool handled)	Unknown
WindowsBase.dll!MS.Win32.HwndSubclass.DispatcherCallbackOperation(object o)	Unknown
WindowsBase.dll!System.Windows.Threading.ExceptionWrapper.InternalRealCall(System.Delegate callback, object args, int numArgs)	Unknown
WindowsBase.dll!System.Windows.Threading.ExceptionWrapper.TryCatchWhen(object source, System.Delegate callback, object args, int numArgs, System.Delegate catchHandler)	Unknown
WindowsBase.dll!System.Windows.Threading.Dispatcher.LegacyInvokeImpl(System.Windows.Threading.DispatcherPriority priority, System.TimeSpan timeout, System.Delegate method, object args, int numArgs)	Unknown
WindowsBase.dll!MS.Win32.HwndSubclass.SubclassWndProc(System.IntPtr hwnd, int msg, System.IntPtr wParam, System.IntPtr lParam)	Unknown
```

**Comment by grokys** [comment]
Fixed by #742.
